comp/trace/agent/component.go

```diff
@@ -0,0 +1,21 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package agent
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+
+	"go.uber.org/fx"
+)
+
+type Component interface {
+	// Run starts the tracer agent.
+	Run()
```

Comment:
Is it possible to not have a `Run` function? The component should be ready once built of use [lifecycle hook](https://uber-go.github.io/fx/lifecycle.html#lifecycle-hooks).